packages/lit-html/src/lit-html.ts

```diff
@@ -637,11 +637,15 @@ class Template {
             // We can't use empty text nodes as markers because they're
             // normalized in some browsers (TODO: check)
             for (let i = 0; i < lastIndex; i++) {
-              (node as Element).append(strings[i] || createMarker());
+              (node as Element).append(strings[i], createMarker());
+              // Walk past the marker node we just added
+              walker.nextNode();
               this.parts.push({type: CHILD_PART, index: ++nodeIndex});
               bindingIndex++;
             }
-            (node as Element).append(strings[lastIndex] || createMarker());
+            // Note the nodeIndex for this marker is accounted for in the
+            // nodeIndex++ expression at the end of the tree walker loop
```

Comment:
```suggestion
            // Note because this marker is added after the walker's current
            // node, it will be walked to (and ignored), so we don't need to
            // adjust nodeIndex here
```